### PR TITLE
OCPBUGS-35744: Fix JSON annotation parsing issue and harden related code

### PR DESCRIPTION
**Pull request by TheRealJon**
Add validation to PackageManifest/CSV annotations that are expected to contain JSON of a certain type (e.g. array of strings, k8s resource, etc.)
Clean up and dedupe code around csv annotations so that all parsing happens in one place and all references to these annotations look alike in the code.

### CODE REVIEW

File: frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-utils.ts [hunk 1]
```diff
@@ -47,36 +50,201 @@ export const isGCPWIFCluster = (
...
+  [OLMAnnotation.CSI]: InfrastructureFeature.CSI,
+  [OLMAnnotation.TLSProfiles]: InfrastructureFeature.TLSProfiles,
+  [OLMAnnotation.TokenAuthAWS]: InfrastructureFeature.TokenAuth,
+  [OLMAnnotation.TokenAuthAzure]: InfrastructureFeature.TokenAuth,
+  [OLMAnnotation.TokenAuthGCP]: InfrastructureFeature.TokenAuthGCP,
+};
+export const normalizeInfrastructureFeature = (feature: string): InfrastructureFeature =>
+  infrastructureFeatureMap[feature] ?? feature;
+
+// TODO Replace with JSONSchema validation
```
Comment (logonoff) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-utils.ts:100]: should there be tickets filed for these TODOs so they're remembered?

Comment (TheRealJon) [reply frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-utils.ts:100]: https://issues.redhat.com/browse/CONSOLE-4281

Comment (logonoff) [reply frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-utils.ts:100]: /lgtm

File: frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx [hunk 2]
```diff
@@ -79,14 +72,16 @@ const ViewInstalledOperatorsButton: React.FC<ViewOperatorButtonProps> = ({ names
 
 const InstallFailedMessage: React.FC<InstallFailedMessageProps> = ({ namespace, csvName, obj }) => {
   const { t } = useTranslation();
+  const hasInitializationResource =
```
Comment (jhadvig) [inline frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx:75]: nit:
```suggestion
  const hasInitializationResourceAnnotation =
```

**Review by jhadvig — APPROVED** [review]
The change looks really good 👍 
Glad that we are adding these validation for JSON schema.
Adding one nit. Otherwise LGTM

/lgtm

**Pull request merged**